monai/handlers/mlflow_handler.py

```diff
@@ -198,4 +313,18 @@ def _default_iteration_log(self, engine: Engine) -> None:
         if not isinstance(loss, dict):
             loss = {self.tag_name: loss.item() if isinstance(loss, torch.Tensor) else loss}
 
-        mlflow.log_metrics(loss, step=engine.state.iteration)
+        if isinstance(engine, Trainer):
+            mlflow.log_metrics(loss, step=engine.state.iteration)
+
+        # If there is optimizer attr in engine, then record parameters specified in init function.
+        try:
+            cur_optimizer = engine.optimizer  # type: ignore
```

Comment:
Please note that not all the MONAI Trainer object has `optimizer` var, let's don't use `try - except` to check it, maybe `hasattr()` is better.
@wyli @ericspod Could you python experts help share some ideas?

Thanks in advance.